### ISSUE TITLE
'There are updates available for this pull request.' - Successfully merged PR is checked out?

### ISSUE THREAD

**Issue by DEberhardt** [microsoft/vscode-pull-request-github#2532]
Hello,

not quite a bug report yet, but odd behaviour I cannot explain. Might be a bug, might be something missing in my knowledge mesh...

```
Version: 1.53.2 (user setup)
Commit: 622cb03f7e070a9670c94bae1a45d78d7181fbd4
Date: 2021-02-11T11:48:04.245Z
Electron: 11.2.1
Chrome: 87.0.4280.141
Node.js: 12.18.3
V8: 8.7.220.31-electron.0
OS: Windows_NT x64 10.0.19042
```

Extension: GitHub Pullrequests

Steps to Reproduce:

Personal - [PR #43 in TeamsFunctions](https://github.com/DEberhardt/TeamsFunctions/pull/43)

1. Starting VS Code
2. Toast: 'There are updates available for this pull request.' received
3. Click on Toast
4. Navigate to Pull Requests
5. Pull Request is checked out in Review Mode
6. There are no updates to this request - the Request was successfully merged on 28-DEC 2020
7. Exit Review Mode
8. Pull request closes
8. Exit VS Code
8. Repeat

I did not check out this request nor do I see any indication that something is off with it... 
This occurred somewhen after PR#45, but cannot pinpoint anything to it.

Help is appreciated.

Thank you,
David

**Comment by RMacfarlane** [comment]
This is interesting - when you "Exit Review Mode" you successfully switch to another branch? If possible, can you copy-paste the output of the `.git/config` file in that repo? I'm wondering if there's an incorrect association between a branch and the pull request

**Comment by DEberhardt** [comment]
yes. it does switch me back to the Main branch, which is also highly confusing as I was starting to tackle some of my TODOs today and inadvertently worked on main (usually always work on branch dev) 
```
[core]
        repositoryformatversion = 0
        filemode = false
        bare = false
        logallrefupdates = true
        symlinks = false
        ignorecase = true
[remote "origin"]
        url = https://github.com/DEberhardt/TeamsFunctions.git
        fetch = +refs/heads/*:refs/remotes/origin/*
[branch "main"]
        remote = origin
        merge = refs/heads/main
[branch "dev"]
        remote = origin
        merge = refs/heads/dev
        github-pr-owner-number = "DEberhardt#TeamsFunctions#43"
```
Here my PR shows up, I wonder what would happen if I were to remove this line?

**Comment by DEberhardt** [comment]
This is now happening again - both `#69` and `#71` are present in my `.git/config` file and won't go away.
While in DEV, Clicking on "Exit Review Mode" on `#69` switches me to MAIN and clicking there on "Exit Review Mode" on `#71` does nothing.

I have not checked out the two commits, yet it insists that they are to be open...

```
[core]
	repositoryformatversion = 0
	filemode = false
	bare = false
	logallrefupdates = true
	symlinks = false
	ignorecase = true
[remote "origin"]
	url = https://github.com/DEberhardt/TeamsFunctions.git
	fetch = +refs/heads/*:refs/remotes/origin/*
[branch "main"]
	remote = origin
	merge = refs/heads/main
	github-pr-owner-number = "DEberhardt#TeamsFunctions#71"
[branch "dev"]
	remote = origin
	merge = refs/heads/dev
	github-pr-owner-number = "DEberhardt#TeamsFunctions#69"
```

I know `#71` I have created today via VsCode and merged it there as well.
No idea why `#69` would show up here.

Anything you want me to try while I have this?

**Comment by alexr00** [comment]
@DEberhardt I think we just don't expect/handle your usual dev flow. Do you always do all your work on the same branch called `dev` then create a PR from `dev` to `main` whenever your work is ready?

**Comment by DEberhardt** [comment]
Hello,
as I am the only contributor to the project currently, yes this is the case.
![image](https://user-images.githubusercontent.com/44290340/121935648-91a45880-cd40-11eb-9383-450c70e57d78.png)
this is my usual MO - work in DEV and merge once every few weeks into MAIN publishing a new version, applying a Tag and creating a release.

This is my first project so I started out with a simple two-branch approach - now that I am more familiar with git, Github & tags, I can revisit this. Still haven't figured out the whole Pipeline yet, so my build script is all run offline as well.

I could not find any significant drawbacks for what I wanted/needed to do, so I have kept it this way. It worked for me so far 😄  
If you have more information, best-practices or other considerations for me, happy to adapt.

**Comment by alexr00** [comment]
@DEberhardt I'm not trying to pass judgement on your dev flow 😅, just trying to explain why this doesn't work in the extension as is. Everything is built around having feature branches that are used for 1 PR and are then not really used again.

I actually think we should support your workflow, especially since the message you get is unclear.

**Comment by DEberhardt** [comment]
no worries. I am a life-long learner, so I am always striving to find new bits for my knowledge mesh 😄
